main/php_globals.h

```diff
@@ -18,6 +18,7 @@
 #define PHP_GLOBALS_H
 
 #include "zend_globals.h"
+#include "php.h"
```

Comment:
Seems unrelated?

Reply:
It's been a while, I think I added this to fix a compilation error I saw when php_globals.h was loaded without loading a header it depended on first.

Reply:
Ah, why is this in php_globals.h at all though? This is Zend functionality, so it should be in zend_globals.h. As is, this causes a layering violation by including PHP headers in Zend.

Reply:
Oh, now I remember.

The issue was that main/php.h includes the `#define PHPAPI`, and this file contains `extern PHPAPI int core_globals_id;
`, so if a C file loaded only main/php_globals.h, it would have a compilation error.

That's out of the scope of this RFC, so I'm taking it out.

Reply:
Just to make sure this didn't get lost in a race condition, did you see my comment on moving this from php_globals.h into zend_globals.h?

Reply:
> Just to make sure this didn't get lost in a race condition, did you see my comment on moving this from php_globals.h into zend_globals.h?

I'm not sure if I understood exactly how you wanted me to implement this, and I'd think that would be more appropriate outside of this PR. It might be easier for you to create a PR/commit than to explain it in that much detail.

Also, that predated my response

> The issue was that main/php.h includes the #define PHPAPI, and this file contains extern PHPAPI int core_globals_id; , so if a C file loaded only main/php_globals.h, it would have a compilation error.

Reply:
Also, I took out the `include "php.h"`

Reply:
Sorry for being unclear. What I meant is to move the declaration of `zend_long exception_string_param_max_len;` from `php_core_globals` into `zend_executor_globals`, which is the same place where `exception_ignore_args` is defined. We should not have dependencies on PHP headers from inside Zend.

Reply:
> Sorry for being unclear. What I meant is to move the declaration of zend_long exception_string_param_max_len; from php_core_globals into zend_executor_globals, which is the same place where exception_ignore_args is defined. We should not have dependencies on PHP headers from inside Zend.


That sounds doable and makes sense

Reply:
> Sorry for being unclear. What I meant is to move the declaration of zend_long exception_string_param_max_len; from php_core_globals into zend_executor_globals, which is the same place where exception_ignore_args is defined. We should not have dependencies on PHP headers from inside Zend.

Okay, done in bcc7adeb5cfc84a1651cd6602e9ca295e39c9658